code/modules/surgery/organs_internal.dm

```diff
@@ -237,19 +265,56 @@
 			"<span class='notice'> You have separated and extracted [target]'s [I] with \the [tool].</span>")
 
 			add_logs(user, target, "surgically removed [I.name] from", addition="INTENT: [uppertext(user.a_intent)]")
-			spread_germs_to_organ(I, user)
+			spread_germs_to_organ(I, user, tool)
 			I.status |= ORGAN_CUT_AWAY
 			var/obj/item/thing = I.remove(target)
 			if(!istype(thing))
 				thing.forceMove(get_turf(target))
 			else
 				user.put_in_hands(thing)
-
-			target.update_icons()
 		else
 			user.visible_message("<span class='notice'>[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!</span>",
 				"<span class='notice'>You can't extract anything from [target]'s [parse_zone(target_zone)]!</span>")
 
+	else if(current_type == "clean")
+		if(!hasorgans(target))
+			return
+		if(!istype(tool,/obj/item/weapon/reagent_containers/))
+			return
+
+		var/obj/item/weapon/reagent_containers/C = tool
+		var/datum/reagents/R = C.reagents
+		var/ethanol = 0 //how much alcohol is in the thing
+
+		if(R.reagent_list.len)
+			for(var/datum/reagent/consumable/ethanol/alcohol in R.reagent_list)
+				ethanol += alcohol.alcohol_perc * 500
+			ethanol /= R.reagent_list.len
+
+		for(var/obj/item/organ/internal/I in affected.internal_organs)
+			if(I)
+				if(R.total_volume < GHETTO_DISINFECT_AMOUNT)
+					user.visible_message("[user] notices there is not enough of \the [tool].", \
+					"You notice there is not enough of \the [tool].")
+					return 0
+				if(I.germ_level < INFECTION_LEVEL_ONE)
+					to_chat(user, "[I] does not appear to be infected.")
+				if(I.germ_level >= INFECTION_LEVEL_ONE)
+					I.surgeryize()//is this even needed?
+					I.germ_level = max(I.germ_level-ethanol, 0)
+					user.visible_message("<span class='notice'> [user] has poured some of \the [tool] over [target]'s [I.name].</span>",
+					"<span class='notice'> You have poured some of \the [tool] over [target]'s [I.name].</span>")
+					R.trans_to(target, GHETTO_DISINFECT_AMOUNT)
+					R.reaction(target, INGEST)
+
+					if(istype(C, /obj/item/weapon/reagent_containers/dropper/) && C.reagents.total_volume<=0) //trans_to() doesnt do this to droppers???
+						var/obj/item/weapon/reagent_containers/dropper/D = C
+						D.filled = 0
+						D.icon_state = "dropper"
+
+				if(R.has_reagent("mitocholide"))
+					affected.status &= ~ORGAN_DEAD
```

Comment:
I'm moving this to it's own surgery thing as per fethas suggestion, so that there is no need to open the ribcage and such before applying mitocholide, which should make it clearer it's only for external parts